numpy/lib/tests/test_function_base.py

```diff
@@ -2817,18 +2817,91 @@ def test_2D(self):
                       [1, 1, 1]])
         assert_array_equal(np.percentile(x, 50, axis=0), [1, 1, 1])
 
-    def test_linear(self):
-
+    def test_linear_inverted_cdf(self):
+        # METHOD 1 of H&F
+        arr = np.asarray([15.0, 20.0, 35.0, 40.0, 50.0])
```

Comment:
```suggestion
    TYPE_CODES = np.typecodes["AllInteger"] + np.typecodes["AllFloat"] + "O"

    @pytest.mark.parametrize("dtype", TYPE_CODES)
    def test_linear_inverted_cdf(self, dtype):
        # METHOD 1 of H&F
        arr = np.asarray([15.0, 20.0, 35.0, 40.0, 50.0], dtype=dtype)
```
It's generally a good idea to check with multiple input dtypes.